library/core/src/stream/stream/next.rs

```diff
@@ -0,0 +1,33 @@
+use crate::future::Future;
+use crate::pin::Pin;
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+/// A future that advances the stream and returns the next value.
+///
+/// This `struct` is created by [`Stream::next`]. See its documentation for more.
+#[unstable(feature = "async_stream", issue = "79024")]
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct Next<'a, S: ?Sized> {
+    stream: &'a mut S,
+}
+
+impl<'a, S: ?Sized> Next<'a, S> {
+    /// Create a new instance of `Next`.
+    pub(crate) fn new(stream: &'a mut S) -> Self {
+        Self { stream }
+    }
+}
+
+#[unstable(feature = "async_stream", issue = "79024")]
+impl<S: Unpin + ?Sized> Unpin for Next<'_, S> {}
+
```

Comment:
This can be removed, as `&'a mut S` is always `Unpin`.
```suggestion
```

Reply:
Ah good catch; not only is this correct -- I also forgot to copy over the impl of `Stream for &mut S`. Fixed both!